### PR TITLE
add looker user creation support during enable lease

### PR DESCRIPTION
**Pull request by sahutd**
Expected req response from looker:

Request1
POST : /users
~~~json
{
    "first_name": "firstname",
    "last_name": "lastname",
    "email": "firstname.lastname@razorpay.com"
}
~~~

Response1
~~~json
{
 "id": 123,
 "is_disabled": false
}
~~~

Request2
POST: /users/:id/credentials_email
~~~json
{
    "email": "firstname.lastname@razorpay.com",
    "type": "email"
}
~~~

Response2
~~~json

~~~

### CODE REVIEW

**Comment by ankitjain28may** [comment]
@sahutd Just a doubt, it doesn't send any response for `Request2` i.e `emailCredentials`? If not, can we use status codes (201 probably)

**Comment by sahutd** [comment]
> @sahutd Just a doubt, it doesn't send any response for `Request2` i.e `emailCredentials`? If not, can we use status codes (201 probably)?

Both the responses send a lot of data. For our use case, those values are not useful. we are already using status codes in the execute request function. anything >= 400 results in an error

fyi this is what is returned

~~~json
{
    "created_at": "2020-10-01T07:54:27.000+00:00",
    "logged_in_at": null,
    "type": "email",
    "email": "",
    "forced_password_reset_at_next_login": false,
    "is_disabled": true,
    "password_reset_url": null,
    "url": "https://localhost:19999/api/3.1/users/951/credentials_email",
    "user_url": "https://localhost:19999/api/3.1/users/951",
    "can": {
        "show_password_reset_url": true
    }
}
~~~

**Review by ccrims0n — COMMENTED** [review]
.

File: ingress_driver/looker.go [hunk 1]
```diff
@@ -127,3 +140,58 @@ func (k *LookerIngressDriver) GetLeaseType() string {
...
+
+	if err != nil {
+		log.Errorf("failed to create looker account for email %s : %s", email, err.Error())
+		return nil, err
+	}
+
+	log.Infof("created looker account for email:%s id:%s", email, resp.Id)
+	log.Infof("creating looker user credentials email for email:%s", email)
+
+	_, createUserCredEmailErr := client.CreateUserCredentialsEmail(resp.Id, pkg.LookerCreateCredentialsEmailRequest{
```
[the remark this answers is not in the document]
Comment (sahutd) [reply ingress_driver/looker.go:171]: 1st call creates a "user entity". 2nd uses user.id returned from 1st to create something called "UserCredentialsEmail". the 2nd call is necessary to "link" the email as a credential to the user id.

File: ingress_driver/looker.go [hunk 2]
```diff
@@ -5,6 +5,7 @@ import (
 	"concierge/pkg"
 	"errors"
 	log "github.com/sirupsen/logrus"
+	"strings"
```
[the remark this answers is not in the document]
Comment (sahutd) [reply ingress_driver/looker.go:8]: ran it. no difference in output

[the remark this answers is not in the document]
Comment (sahutd) [reply ingress_driver/looker.go:8]: ~/concierge   lookerUserCreateDuringLeaseCreate  go fmt ingress_driver/looker.go                                                                                                                                                 ✔
 ~/concierge   lookerUserCreateDuringLeaseCreate                                                                                                                                                                                  ✔

File: ingress_driver/looker.go [hunk 3]
```diff
@@ -127,3 +140,58 @@ func (k *LookerIngressDriver) GetLeaseType() string {
...
+
+	if createUserCredEmailErr != nil {
+		log.Errorf("failed to create looker user credentials email for email %s : %s", email, createUserCredEmailErr.Error())
+		return nil, createUserCredEmailErr
+	}
+
+	return resp, err
+}
+
+func parseFirstAndLastNameFromEmail(email string) (string, string, error) {
```
[the remark this answers is not in the document]
Comment (sahutd) [reply ingress_driver/looker.go:184]: no. in users table we are storing this way

| id | name               | username           | email                           | email_verified_at | admin  | google_id | access_token | password | remember_token | deleted_at | created_at          | updated_at          |
+----+--------------------+--------------------+---------------------------------+-------------------+--------+-----------+--------------+----------+----------------+------------+---------------------+---------------------+
| 2  | saimadhav.heblikar | saimadhav.heblikar | saimadhav.heblikar@razorpay.com | <null>            | 1      |           |              | <null>   | <null>         | <null>     | 2020-09-11 16:27:26 | 2020-10-15 02:19:45

File: ingress_driver/looker.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (sahutd) [reply ingress_driver/looker.go:171]: without the 2nd call, there was no mention of the email id on any of the attributes/pages related to the user

**Pull request closed without merging**